### ISSUE TITLE
Bad fs structure with `crystal init lib /path/to/project`

### ISSUE THREAD

**Issue by bew** [crystal-lang/crystal#4817]
While working on #4691 I've found a weird behavior of some specs:
https://github.com/crystal-lang/crystal/blob/b5dd948dfb74e883e64156a121f3f1d871cd2c3e/spec/compiler/crystal/tools/init_spec.cr#L166

As an example it'll generate the following project structure:
```
$ crystal init lib /tmp/path/to/my_lib
      create  /tmp/path/to/my_lib/.gitignore
      create  /tmp/path/to/my_lib/.editorconfig
      create  /tmp/path/to/my_lib/LICENSE
      create  /tmp/path/to/my_lib/README.md
      create  /tmp/path/to/my_lib/.travis.yml
      create  /tmp/path/to/my_lib/shard.yml
      create  /tmp/path/to/my_lib/src//tmp/path/to/my_lib.cr
      create  /tmp/path/to/my_lib/src//tmp/path/to/my_lib/version.cr
      create  /tmp/path/to/my_lib/spec/spec_helper.cr
      create  /tmp/path/to/my_lib/spec//tmp/path/to/my_lib_spec.cr
Initialized empty Git repository in /tmp/path/to/my_lib/.git/
$ tree /tmp/path/to/my_lib
/tmp/path/to/my_lib/
├── spec/
│   ├── tmp/
│   │   └── path/
│   │       └── to/
│   │           └── my_lib_spec.cr
│   └── spec_helper.cr
├── src/
│   └── tmp/
│       └── path/
│           └── to/
│               ├── my_lib/
│               │   └── version.cr
│               └── my_lib.cr
├── LICENSE
├── README.md
└── shard.yml
```
We can see that there are some files with incorrect path, like `/tmp/path/to/my_lib/src//tmp/path/to/my_lib.cr`.

Here using `/tmp/path/to/my_lib` will use `/tmp/path/to/` as the path, and `/tmp/path/to/my_lib` as the project name, and some files are generated as `<path>/src/<project_name>/file.cr`.

---

I think this is an unwanted behavior, there are multiple things we could do to prevent it:

1. **Prevent the use of `/` in the project name**: we ensure the following usage `crystal init lib|app <project_name> <path>`. An example would be `crystal init lib my_lib /tmp/path/to/my_lib`. The only drawback I see is that we need to write the project name twice, if we want it to be in a specific path.

2. **Extract the project name from the given path**: `/tmp/path/to/my_lib` will give `/tmp/path/to/` as path, and `my_lib` as the project name. The example `crystal init lib /tmp/path/to/my_lib` would then work as expected.

What do you think?

**Comment by RX14** [comment]
I think the only bug here is that `/` and `\` are allowed in names. It should be a name, not a path.

**Comment by bew** [comment]
@RX14 I agree, so an example would be `crystal init lib my_lib /tmp/path/to/my_lib`.
The only drawback I see is that we need to write the project name twice, if we want it to be in a specific path.

**Comment by RX14** [comment]
That's not a problem, really.

**Comment by straight-shoota** [comment]
I guess the name could be extracted if a path is given... `crystal init lib /tmp/path/to/my_lib` means it is initialized in folder `/tmp/path/to/my_lib` and named `my_lib`.

**Comment by bew** [comment]
Actually the generated initial code is broken as well:
```cr
# file src/tmp/path/to/my_lib.cr
require ".//tmp/path/to/my_lib/*"

# TODO: Write documentation for `/tmp/path/to/myLib`
module /tmp/path/to/myLib
  # TODO: Put your code here
end
```
```cr
# file spec/tmp/path/to/my_lib_spec.cr
require "./spec_helper"

describe /tmp/path/to/myLib do
  # TODO: Write tests

  it "works" do
    false.should eq(true)
  end
end
```

**Comment by straight-shoota** [comment]
This should be fixed by #8737
